src/shared/Core/X509Utils.cs

```diff
@@ -0,0 +1,23 @@
+using System.Security.Cryptography.X509Certificates;
+
+namespace GitCredentialManager;
+
+public static class X509Utils
+{
+    public static X509Certificate2 GetCertificateByThumbprint(string thumbprint)
```

Comment:
Here's a slight more battle tested way to load certs. I think this is fine for loading by thumbprint, but if loading by name you should exclude expired certs and load the most fresh.

https://github.com/AzureAD/microsoft-identity-web/blob/96323e40bc6e6610192461f7b1dfaf4c2605ff21/src/Microsoft.Identity.Web.Certificate/CertificateLoaderHelper.cs#L55

Reply:
Is there a compelling argument for allowing people to specify something other than a thumbprint (like the cert name, or file path)? Are there common situations where the thumbprint would not be known?

I guess at certificate renewal a new thumbprint would be created, but the certificate would have the same name?

Right now we only have `credential.azreposServicePrincipalCertificateThumbprint` / `GCM_AZREPOS_SP_CERT_THUMBPRINT` to specify a thumbprint.

We could add other options like `..CertificatePath` / `.._CERT_PATH` and `..CertificateName` / `.._CERT_NAME` in the future (unless you think it's useful from the get go!) :)